### ISSUE TITLE
Globe Lights unavailable after SmartLife App device update

### ISSUE THREAD

**Issue by JAudi23** [rospogrigio/localtuya#1490]
I have three Globe brand smart lights that were working fine using localtuya, when I accepted updates (Main Module and MCU module v1.5.21) to two of the devices via the SmartLife app.
After that update, the devices and entities for those two bulbs appeared Unavailable in Home Assistant, but worked fine via the SmartLife app.
When attempting to edit the existing devices in localtuya, they all appeared in the list, but still showed unavailable.

I tried to remove all three of my devices from the localtuya integration deleted the integration, restarted HA, then reinstalled the integration.
Now only the one bulb that I did not update in the smartlife app is visible and added back to the integration.  I do not see the two updated lights when I try to add them, although in my Tuya IoT account, all three are visible and showing as online.

Any suggestions to point me in the right direction or did I kill these bulbs ability to be used in HA now?

**Comment by xZetsubou** [comment]
I will assume that these updated bulbs changed protocol, what do you mean by
> I do not see the two updated lights when I try to add them

Like they don't show up in discovered list? 
One thing that people might misunderstand, discovery works without cloud it pull the local devices in network fact the shows in cloud doesn't means then need to show in discovery as well.

**Comment by JAudi23** [comment]
Sorry for not clarifying, I am referring to the configure dialog to add a new device as shown in the below screenshot.

![image](https://github.com/rospogrigio/localtuya/assets/107599616/ea24c3d6-f368-4666-b50f-4e828fd1e143)

Prior to the device update I would see all 3 devices here.  As you can see they are not found/discovered.  I was able to re-add the one light after reinstalling the integration that I did not update via the app.

![image](https://github.com/rospogrigio/localtuya/assets/107599616/7abe2817-3802-4703-b341-463d1f382a80)

One other thing I noticed when I scan my local network with a tool I use, is the light that can be discovered in HA is showing a name of "wlan0" whereas the other two that are not discovered in HA are showing "lwip0".  I'm not sure what this refers to?

![image](https://github.com/rospogrigio/localtuya/assets/107599616/1a8f3d31-d856-49de-9c92-09b537d952b3)

**Comment by xZetsubou** [comment]
I can guess what can cause this issue, rather then explain it, You can [test my fork ](https://github.com/xZetsubou/hass-localtuya)and see if your issue got fixed if it did I can. explain more.

**Comment by JAudi23** [comment]
Can both integrations run simultaneously, the original and your fork, or do I need to fully remove the original integrations?

**Comment by JAudi23** [comment]
In any case, I've completely removed the original integration, restarted HA, and installed your fork.
After clicking "configure" I get this blank dialog which I assume is the "Add new" " Edit Existing", etc options.
I'll try in light mode instead of dark mode to see if it's related.

![image](https://github.com/rospogrigio/localtuya/assets/107599616/9fa63740-31a3-440a-a95d-bf21c74531da)

**Comment by JAudi23** [comment]
Same issue, will try another restart of HA
![image](https://github.com/rospogrigio/localtuya/assets/107599616/5c92c519-5ddc-4ead-9336-fa89bd3321ec)

**Comment by JAudi23** [comment]
Restarting HA did not make a difference.
I selected the first option (which still remained blank), and it did provide the "add new entry" dialog.
However, the dropdown still only lists the single device which I did not update through the Smart Life app.
In case it matters I chose to install your beta version of the integration (v3.1.0b4)

![image](https://github.com/rospogrigio/localtuya/assets/107599616/87bc2440-a864-455c-9c70-9dac2eb9043f)

**Comment by xZetsubou** [comment]
Yes it seems 3.5 devices doesn't get discover, can you try to add it manually as `3.5` as for how to get the infos 
Download the txt file and search for the devices and try to insert manually if it didn't connect it seems `3.5` still bugged, I'm search for `3.5` device to debug this issue

![image](https://github.com/rospogrigio/localtuya/assets/46300268/584a6500-9e0c-4cfc-a6ca-b569a75eb5d3)

**Comment by JAudi23** [comment]
I do not recall how to find the devices local key.  can you help guide me?

**Comment by JAudi23** [comment]
Disregard the above, I found the local key via the IoT Development portal, however it still does not add manually.
Any suggestions?

**Comment by JAudi23** [comment]
The error message I receive after trying to add a device manually is below.  I tried to both add using "auto" protocol, as well as "3.5"
![image](https://github.com/rospogrigio/localtuya/assets/107599616/3e0684f6-4451-4510-841b-28ffed248f65)

**Comment by xZetsubou** [comment]
> Disregard the above, I found the local key via the IoT Development portal, however it still does not add manually. Any suggestions?

I'm digging into this issue it seems `3.5` doesn't connect successfully, It's hard to debug without device I order a device should arrive today hoping it's `3.5`, I can make a quick change if you don't mind to test it.

**Comment by JAudi23** [comment]
I don't mind testing as currently they are useless to me in Home Assistant.
If I understand, when I updated the firmware through the Smart Life App, it upgraded the protocol to version 3.5?
I was successfully using them on protocol v3.4 previous to the update.

**Comment by xZetsubou** [comment]
Yes, if you want to make sure even more use TinyTuya scanner with install there library and test.
I released [beta version](https://github.com/xZetsubou/hass-localtuya/releases/tag/3.1.0b_protocol_35b1) test it out.
If there it didn't works open same issue in my fork or discussion since this issue have nothing to do with this repo.

Check this comment TinyTuya scan: https://github.com/xZetsubou/hass-localtuya/issues/25#issuecomment-1712754048

**Comment by JAudi23** [comment]
I've updated to the latest beta, but need to restart HA, and will do that in the morning to test it out.

**Comment by xZetsubou** [comment]
Alright thanks, Either way I need to wait for my device with hope that it's `3.5` and might need to fix more stuff.

**Comment by JAudi23** [comment]
I ran TinyTuya as per your comment, and you are correct, the device that is working is still 3.4, and the other two which no longer work, are 3.5.

**Comment by xZetsubou** [comment]
Update: I pushed newer version beta2.

**Comment by xZetsubou** [comment]
Yeah, I kinda figured out how to fix this but instead of push more beta updates atm, I am receiving the device that I ordered probably today hoping it's updated `3.5`.

**Comment by sidevesh** [comment]
Facing this same issue recently with a new unit of the same exact bulb that I have a bunch of,
tried updating this, @xZetsubou 's fork including the 3.5 beta, same issue.
Had to switch to Tuya official integration that depends on cloud for now :(

**Comment by xZetsubou** [comment]
I literally bought +4 devices that looks similar to `3.5` devices I see but most of them are `3.4` , it possible to fix the issue without having device to debug but it might takes time.

@sidevesh If I maybe ask where did you bought that bulb from is it available in global amazon? 
Can you tell me the what are `Main Version` and `MCU Version` in smart life app of that bulb?


I ordered 2 more bulb it will arrive today as well hoping that one of them are `3.5`

**Comment by sidevesh** [comment]
Main Module: V1.5.21
MCU Module: V1.5.21

This is an Indian brand selling these rebranded, https://www.amazon.in/gp/product/B08DLJSGJ5/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1

I have gone through so many of these bulbs that I think I have seen this same bulb go from 3.3 to 3.4 to 3.5 now 🤣 
Exact same bulb, exact same Amazon listing

**Comment by JAudi23** [comment]
> Facing this same issue recently with a new unit of the same exact bulb that I have a bunch of, tried updating this, @xZetsubou 's fork including the 3.5 beta, same issue. Had to switch to Tuya official integration that depends on cloud for now :(

does the official Tuya integration in HA still work with these devices that are protocol version 3.5?
I didn't think of that, but might be my go to as well.  I'll hold out a bit longer for a workaround here though.

**Comment by xZetsubou** [comment]
Good news, I finally got `3.5` device after buying 7 bulbs.

**Comment by sidevesh** [comment]
Latest @xZetsubou 's fork worked for me to get these 3.5 bulbs working locally again

**Comment by lluisd** [comment]
Here with the same problem. My new socket (which is the same model as my old sockets but newer firmware) shows up in my router as lwip0 instead of wlan0 so my localtuya cannot detect it. I will try what you said.
